ui/packages/atlasmap/src/Atlasmap/utils/ui.ts

```diff
@@ -252,6 +252,10 @@ export function executeFieldSearch(
   );
 }
 
+export function mappingHasSourceCollection(): boolean {
+  return initializationService.cfg.mappingService.willClearOutSourceFieldsOnTogglingExpression();
```

Comment:
This feels like it could be resolved to be a boolean returned by the `useAtlasmap` hook, reducing even further the need for the ui components to know about how the core works

Reply:
done.